File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.java
class MyQueue {

    Stack<Integer> stack1;
    Stack<Integer> stack2;
    public MyQueue() {
        stack1 = new  Stack<Integer>();
        stack2 = new Stack<Integer>();
    }
    
    public void push(int x) {
        stack1.push(x);
    }
    
    public int pop() {
        if(stack2.size() > 0) {
            return stack2.pop();
        } else {
            while(stack1.size() > 0) {
                stack2.push(stack1.pop());
            }
            return stack2.pop();
        }
    }
    
    public int peek() {
        if(stack2.size() > 0) {
            return stack2.peek();
        } else {
            while(stack1.size() > 0) {
                stack2.push(stack1.pop());
            }
            return stack2.peek();
        }
    }
    
    public boolean empty() {
        return !(stack1.size() > 0 || stack2.size() > 0);
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */

File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.js

var MyQueue = function() {
    this.stack = [];
    this.start = -1;
    this.end = -1;
};

/** 
 * @param {number} x
 * @return {void}
 */
MyQueue.prototype.push = function(x) {
    this.stack.push(x);
    if(this.start < 0) this.start++;
    this.end++;
};

/**
 * @return {number}
 */
MyQueue.prototype.pop = function() {
    return this.stack[this.start++];
};

/**
 * @return {number}
 */
MyQueue.prototype.peek = function() {
    return this.stack[this.start];
};

/**
 * @return {boolean}
 */
MyQueue.prototype.empty = function() {
    if(this.start > this.end || this.start == -1) return true;
    return false;
};

/** 
 * Your MyQueue object will be instantiated and called as such:
 * var obj = new MyQueue()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.peek()
 * var param_4 = obj.empty()
 */